src/lib/assistantStats/refresh-assistants-counts.ts

```diff
@@ -0,0 +1,119 @@
+import type { Migration } from "../migrations/routines";
+import { client, collections, getCollections } from "$lib/server/database";
+import { ObjectId } from "mongodb";
+import type { MigrationResult } from "$lib/types/MigrationResult";
+import { acquireLock, isDBLocked, refreshLock, releaseLock } from "$lib/migrations/lock";
+
+const migration: Migration = {
+	_id: new ObjectId("65f7ff14298f30c5060eb6ac"),
+	name: "Refresh assistants count for last 24 hours",
+	// eslint-disable-next-line no-shadow
+	up: async (client) => {
+		const { assistantStats } = getCollections(client);
+		const currentDate = new Date();
+		const twentyFourHoursAgo = new Date();
+		twentyFourHoursAgo.setDate(twentyFourHoursAgo.getDate() - 1); // subtract 24 hours (i.e. 1 day)
+
+		assistantStats.aggregate([
+			{ $match: { dateDay: { $gte: twentyFourHoursAgo, $lt: currentDate } } },
+			{
+				$group: {
+					_id: { _id: "$_id", dateWithHour: "$dateWithHour" },
+					totalCount: { $sum: "$count" },
+				},
+			},
+			{
+				$merge: {
+					into: "assistants",
+					on: "_id",
+					whenMatched: [
+						{
+							$set: {
+								"last24HoursCount.count": "$$ROOT.totalCount",
```

Comment:
I don't think you need to put it in a `Migration` anymore?

Reply:
I was in assumption that multiple machines are running chat-ui (and the call below might be called my multiple machines within a same period). therefore, we would need to use Migration & locks. No ?

https://github.com/huggingface/chat-ui/blob/58f3eace2bf698132be68d5a69b2f61b88a0111f/src/hooks.server.ts#L26

Reply:
locks yes, but no need for `Migration`

We can create a lock on `refresh-assistant-counts`

Reply:
handled in 08037405fa7c562f68501ebbfb4b1150f8e5a4ae

1. Removed usage of `Migration`
2. Also set those lines below to make sure that different machines will try to run the refresh "cron" job at the same time (i.e. synced even if the machines start at different times)

https://github.com/huggingface/chat-ui/blob/08037405fa7c562f68501ebbfb4b1150f8e5a4ae/src/lib/assistantStats/refresh-assistants-counts.ts#L70-L79